mycroft/gui/extensions.py

```diff
@@ -0,0 +1,243 @@
+import threading
+from mycroft.configuration import Configuration
+from mycroft.messagebus import Message
+from mycroft.util.log import LOG
+from mycroft.api import is_paired
+from ovos_utils.system import ssh_enable, ssh_disable
+
+from json_database import JsonStorageXDG
+from mycroft.gui.homescreen import HomescreenManager
+
+
+class ExtensionsManager():
+    def __init__(self, name, bus, gui):
+        """ Constructor for the Extension Manager. The Extension Manager is responsible for
+        managing the extensions that define additional GUI behaviours for specific platforms.
+
+        Args:
+            name: Name of the extension manager
+            bus: MessageBus instance
+            gui: GUI instance
+        """
+
+        self.name = name
+        self.bus = bus
+        self.gui = gui
+        config = Configuration.get()
+        enclosure_config = config.get("gui")
+        self.active_extension = enclosure_config.get("extension", "Generic")
+
+        # ToDo: Add Exclusive Support For "Desktop", "Mobile" Extensions
+        self.supported_extensions = ["SmartSpeaker", "Bigscreen", "Generic"]
```

Comment:
please make lower case, want it to work for "generic", "GENERIC", "Generic" and "GeNeRiC"   :P